### PR TITLE
Convert Identity to Reference

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Part of #4246.

### CODE REVIEW

File: build/repo.props [hunk 1]
```diff
@@ -66,6 +66,10 @@
                           "
                           Condition=" '$(MSBuildRuntimeType)' == 'Core' " />
 
+
+    <!-- Exclude the WPF samples for now because they use classic .csproj, which is not yet supported in our build. -->
+    <ProjectToExclude Include="$(RepositoryRoot)src\Identity\Identity\samples\NativeWPFClient\NativeWPFClient.csproj;" />
```
Comment (JunTaoLuo) [inline build/repo.props]: This doesn't seem to work when I try to build the solution with dotnet build Identity.sln. Is that expected?

**Comment by JunTaoLuo** [comment]
WIP since I broke tests.

File: build/repo.props (hunk 1 above)
Comment (natemcmaster) [reply build/repo.props]: yes, that's expected. You would have seen the same problem before unless you did `dotnet build -c DebugNoWPF Identity.sln`

File: build/repo.props [hunk 2]
```diff
@@ -83,6 +87,7 @@
                           $(RepositoryRoot)src\Hosting\**\*.*proj;
                           $(RepositoryRoot)src\Http\**\*.*proj;
                           $(RepositoryRoot)src\Html\**\*.*proj;
+                          $(RepositoryRoot)Identity\**\*.*proj;
```
Comment (natemcmaster) [inline build/repo.props]:
```suggestion
                          $(RepositoryRoot)src\Identity\**\*.*proj;
```

File: src/Identity/README.md [hunk 3]
```diff
@@ -1,30 +0,0 @@
-ASP.NET Core Identity
```
Comment (natemcmaster) [inline src/Identity/README.md]: Can we keep this file? The list of community providers is still useful.

**Review by natemcmaster — APPROVED** [review]
Only saw one other thing. Otherwise this looks fine.

**Pull request merged**